src/Store/Bridge/MariaDB/Store.php

```diff
@@ -0,0 +1,153 @@
+
+            $statement->execute($operation);
+        }
+    }
+
+    /**
+     * @param array{
+     *     limit?: positive-int,
+     * } $options
+     */
+    public function query(Vector $vector, array $options = [], ?float $minScore = null): array
+    {
+        $statement = $this->connection->prepare(
+            \sprintf(
+                <<<'SQL'
+                    SELECT id, VEC_ToText(%1$s) embedding, metadata, VEC_DISTANCE_EUCLIDEAN(%1$s, VEC_FromText(:embedding)) AS score
+                    FROM %2$s.%3$s
+                    %4$s
+                    ORDER BY score ASC
+                    LIMIT %5$d
+                    SQL,
+                $this->vectorFieldName,
+                $this->databaseName,
+                $this->tableName,
+                null !== $minScore ? 'WHERE VEC_DISTANCE_EUCLIDEAN(%1$s, VEC_FromText(:embedding)) >= :minScore' : '',
+                $options['limit'] ?? 5,
+            ),
+        );
+
+        $params = ['embedding' => json_encode($vector->getData())];
+
+        if (null !== $minScore) {
+            $params['minScore'] = $minScore;
+        }
+
+        $documents = [];
+
+        $statement->execute($params);
+
+        foreach ($statement->fetchAll(\PDO::FETCH_ASSOC) as $result) {
+            $documents[] = new VectorDocument(
+                id: $this->toUuid($result['id']),
+                vector: new Vector(json_decode((string) $result['embedding'], true)),
+                metadata: new Metadata(json_decode($result['metadata'] ?? '{}', true)),
+                score: $result['score'],
+            );
+        }
+
+        return $documents;
+    }
+
+    /**
+     * @param array{} $options
+     */
+    public function initialize(array $options = []): void
+    {
+        if ([] !== $options) {
+            throw new InvalidArgumentException('No supported options');
+        }
+
+        $serverVersion = $this->connection->getAttribute(\PDO::ATTR_SERVER_VERSION);
+
+        if (!str_contains((string) $serverVersion, 'MariaDB') || version_compare($serverVersion, '11.7.0') < 0) {
+            throw new InvalidArgumentException('You need MariaDB >=11.7 to use this feature');
+        }
+
+        $this->connection->exec(
+            \sprintf(
+                <<<'SQL'
+                    CREATE TABLE IF NOT EXISTS %1$s.%2$s (
+                        id BINARY(16) NOT NULL PRIMARY KEY,
+                        metadata JSON,
+                        %3$s VECTOR(1536) NOT NULL,
+                        VECTOR INDEX %4$s (%3$s)
+                    )
+                    SQL,
+                $this->databaseName,
+                $this->tableName,
+                $this->vectorFieldName,
+                $this->indexName,
+            ),
+        );
+    }
+
+    private function toBinary(Uuid $uuid): string
+    {
+        return $uuid->toBinary();
+    }
+
+    private function toUuid(string $binary): Uuid
+    {
+        return Uuid::fromString($binary);
+    }
```

Comment:
what's the purpose of those methods? we could just inline that, can't we?

Reply:
Sure – I used MongoDB store as a template and it had these

I'll inline